### PR TITLE
Refactor scheduler integration test that starts APIServer in a non-standarized manner

### PR DESCRIPTION
**Pull request by Huang-Wei**
#### What type of PR is this?

/kind cleanup
/kind failing-test
/sig scheduling

#### What this PR does / why we need it:

Follow-up of https://github.com/kubernetes/kubernetes/issues/109783#issuecomment-1120465908.

Due to some legacy issues, the scheduler integration test spins up its own API server. As time goes, it diverges with the standardized way of initializing an APIServer instance (APIPriorityAndFairness, Flowcontrol, etc.)

This PR migrates to use the standardized way to initialize an APIServer (`RunAnAPIServer()`), and also change the unlimited QPS to 100.

#### Which issue(s) this PR fixes:

Related with #109783 and flaky scheduler integration test.

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: test/integration/util/util.go [hunk 1]
```diff
@@ -350,29 +350,25 @@ func InitTestAPIServer(t *testing.T, nsPrefix string, admission admission.Interf
...
+		testCtx.NS = framework.CreateTestingNamespace(nsPrefix+string(uuid.NewUUID()), testCtx.HTTPServer, t)
 	} else {
-		testCtx.NS = framework.CreateTestingNamespace("default", s, t)
+		testCtx.NS = framework.CreateTestingNamespace("default", testCtx.HTTPServer, t)
 	}
 
 	// 2. Create kubeclient
 	kubeConfig := &restclient.Config{
-		QPS: -1, Host: s.URL,
+		QPS:   100,
```
Comment (Huang-Wei) [inline test/integration/util/util.go:369]: The scheduler perf test uses [300](https://github.com/kubernetes/test-infra/blob/5f20e82ebae61f5c00247fde5386c435cf372257/config/jobs/kubernetes/sig-scalability/sig-scalability-release-blocking-jobs.yaml#L117).

In terms of regular integration tests, 100 should suffice.

**Comment by aojea** [comment]
+1

**Pull request merged**